### PR TITLE
docs: custom 404 page

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

adds a custom 404 page to the documentation

## Checklist
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/404.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. SPDX-License-Identifier: MIT
+
+:orphan:
```
Comment (onerandomusername) [inline docs/404.rst:3]: the contents of this page could be improved upon, open for suggestions.

File: docs/conf.py [hunk 2]
```diff
@@ -50,6 +50,7 @@
     "attributetable",
     "resourcelinks",
     "nitpick_file_ignorer",
+    "notfound.extension",
```
Comment (shiftinv) [inline docs/conf.py:53]: This should be moved below `hoverxref.extension`, so that all custom extensions are at the end

Comment (onerandomusername) [reply docs/conf.py]: https://github.com/DisnakeDev/disnake/pull/797/commits/46352b863aebe7bb590c570cda56ae9702cb4ad2

**Comment by shiftinv** [comment]
On medium size screens, this looks slightly weird as there's still the margin for the (non-existent) sidebar:
image

This doesn't happen below 1050px (separate menu for sidebar) or above 1500px (you start also getting a margin on the right, which makes this look fine).

I suppose we'd either have to put *something* in the sidebar, or adjust the css to center the content a bit better.
But then again, the same thing currently happens on the index pages, so we might want to leave it as is for now.

File: docs/404.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+We could not find what you were looking for.
+
+
+You might find what you're looking for in one of the below pages:
+
+.. toctree::
+    :maxdepth: 1
+    :titlesonly:
+
+    Index <index>
```
Comment (shiftinv) [inline docs/404.rst:18]: These don't link correctly on non-top-level pages, e.g.
https://disnake--797.org.readthedocs.build/en/797/ext/commands/a/b/c/d/stuff

Might be a bug with the extension not updating references in the toc correctly? Not sure.

**Comment by onerandomusername** [comment]
> But then again, the same thing currently happens on the index pages, so we might want to leave it as is for now.

Since this already occurs I think this is a non-issue/out of scope.

File: docs/404.rst (hunk 3 above)
Comment (onerandomusername) [reply docs/404.rst]: limitation, and not easily solvable. Completely changed the contents in https://github.com/DisnakeDev/disnake/pull/797/commits/61ec6d8592c3ee6536e89a2e03fd1fddfd8d09d6 to make this a nonissue.

File: changelog/797.doc.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Added a custom 404 page for when the navigated page does not exist.
```
Comment (shiftinv) [inline changelog/797.doc.rst:1]:
```suggestion
Add a custom 404 page for when the navigated page does not exist.
```

**Review by shiftinv — APPROVED** [review]
other than that, lgtm

**Pull request merged**